### PR TITLE
ENH: fix export macros usage to match ITK rules

### PR DESCRIPTION
**Pull request by laurentmalka**
- The rules are the following:
    - Use ModuleName_EXPORT for non-templated classes.
    - Use ITK_TEMPLATE_EXPORT for templated classes.
    - Use ITK_FORWARD_EXPORT for forward declarations.

### CODE REVIEW

**Comment by laurentmalka** [comment]
@LucasGandel

**Comment by SimonRit** [comment]
Thanks, for the clarification of the rules and for the code. Are the rules explained somewhere in the ITK software guide? A link would be useful, it's always been a bit of a mistery to me (also because these exports are useless on Linux).
The change looks good but there are now new warnings:
https://my.cdash.org/viewBuildError.php?type=1&buildid=1954275
I think they should be addressed in a separate commit. Do you mind doing it in the same PR?

**Review by SimonRit — APPROVED** [review]

**Comment by LucasGandel** [comment]
Thanks @SimonRit !

Laurent saw the warning and is working on fixing it, I'll review as soon as it is pushed.

The original discussion about export can be found [here](https://discourse.itk.org/t/correct-usage-of-export-macros/1304), maybe we can describe this in some kind of RTK contribution/developper guide. file?

**Comment by SimonRit** [comment]
Thanks, I now see it's already page 364 of the [ItkSoftwareGuide](https://itk.org/ItkSoftwareGuide.pdf), no need to repeat this. The rule is still to follow ITK guidelines in RTK, thus minimizing the RTK developer guide...

**Comment by SimonRit** [comment]
Thanks it works! But I don't like so much the change, we should not have to inline functions for this reason. Have you tried including [itkWin32Header.h](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkWin32Header.h)? It seems to simply disable the warning...

**Comment by laurentmalka** [comment]
I misunderstood what the warning meant exactly, but you are right, disabling the warning should be perfectly fine as long as the member variables using STL types remain as private or protected fields.  
This should now be fixed.

I noticed the pipeline for my last push failed though, and it seems to have been caused by a timeout. Is there something I need to fix or is that just something that happens sometimes ?

**Comment by SimonRit** [comment]
I think it's still running isn't it?
https://github.com/SimonRit/RTK/actions/runs/503159488
You'll have to change the commit name with a `COMP:` prefix (see [CONTRIBUTING.md](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md)). Can you add the warnings (or one example) in the commit message for future reference? Thanks again

**Comment by laurentmalka** [comment]
I was talking about this one, which failed  
https://github.com/SimonRit/RTK/actions/runs/501139953

**Comment by SimonRit** [comment]
> I was talking about this one, which failed
> https://github.com/SimonRit/RTK/actions/runs/501139953

Ok thanks. Let's see if it happens to the latest commit. We might have to split the MacOS compilations, but I'd like to propagate such a change to ITKModuleTemplate for consistency.

File: include/rtkCudaScatterGlareCorrectionImageFilter.h [hunk 1]
```diff
@@ -26,6 +26,7 @@
 
 #  include "rtkScatterGlareCorrectionImageFilter.h"
 #  include "rtkCudaFFTProjectionsConvolutionImageFilter.h"
+#  include "RTKExport.h"
```
Comment (LucasGandel) [inline include/rtkCudaScatterGlareCorrectionImageFilter.h]: Unneeded?

File: include/rtkDualEnergyNegativeLogLikelihood.h [hunk 2]
```diff
@@ -20,6 +20,7 @@
 #define rtkDualEnergyNegativeLogLikelihood_h
 
 #include "rtkProjectionsDecompositionNegativeLogLikelihood.h"
+#include "RTKExport.h"
```
Comment (LucasGandel) [inline include/rtkDualEnergyNegativeLogLikelihood.h]: Unneeded?

File: include/rtkCudaScatterGlareCorrectionImageFilter.h (hunk 1 above)
Comment (LucasGandel) [inline include/rtkCudaScatterGlareCorrectionImageFilter.h]: Unneeded?

File: include/rtkDualEnergyNegativeLogLikelihood.h (hunk 2 above)
Comment (LucasGandel) [inline include/rtkDualEnergyNegativeLogLikelihood.h]: Unneeded?

File: include/rtkProjectionsDecompositionNegativeLogLikelihood.h [hunk 3]
```diff
@@ -23,6 +23,7 @@
 #include <itkVectorImage.h>
 #include <itkVariableLengthVector.h>
 #include <itkVariableSizeMatrix.h>
+#include "RTKExport.h"
```
Comment (LucasGandel) [inline include/rtkProjectionsDecompositionNegativeLogLikelihood.h]: Unneeded?

File: include/rtkSchlomka2008NegativeLogLikelihood.h [hunk 4]
```diff
@@ -20,6 +20,7 @@
 #define rtkSchlomka2008NegativeLogLikelihood_h
 
 #include "rtkProjectionsDecompositionNegativeLogLikelihood.h"
+#include "RTKExport.h"
```
Comment (LucasGandel) [inline include/rtkSchlomka2008NegativeLogLikelihood.h]: Unneeded?

**Review by LucasGandel — APPROVED** [review]
@laurentmalka Thanks! The changes look good!
I added a few "Unneeded?" comments in classes that include rtkExport.h but don't use RTK_EXPORT. Did you forget to call RTK_EXPORT for those classes or is the include unneeded?

**Comment by SimonRit** [comment]
Thanks again!

**Pull request merged**